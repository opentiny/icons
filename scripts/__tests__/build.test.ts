import { test, expect } from 'vitest'
import { genClass, loadSvg, parsePath } from '../build'

test('测试解析路径的正确性', () => {
  const svgPath = './svgs/base/1-方向/1#前进-go.svg'
  const { category, group, nameCn, name, order } = parsePath(svgPath)

  expect(category).toEqual('base')
  expect(group).toEqual('1-方向')
  expect(nameCn).toEqual('前进')
  expect(name).toEqual('go')
  expect(order).toEqual(1)
})

test('测试生成css的正确性', () => {
  let svg = loadSvg('./scripts/__tests__/点击-click.svg')
  let css = genClass('click', svg.toMinifiedString())

  expect(css).toMatchSnapshot()

  svg = loadSvg('./scripts/__tests__/函数工作流-FunctionGraph.svg')
  css = genClass('fg', svg.toMinifiedString())

  expect(css).toMatchSnapshot()
})
