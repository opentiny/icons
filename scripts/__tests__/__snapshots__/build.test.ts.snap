// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`测试生成css的正确性 1`] = `".ci-click{--v:url("data:image/svg+xml;utf8,%3Csvg width='16.000000' height='16.000000' viewBox='0 0 16 16' xmlns='http://www.w3.org/2000/svg'%3E%3Cg fill='none'%3E%3Cg clip-path='url(%23prefix__clip15_952)'%3E%3Cpath id='prefix__&%23x5f62;&%23x72b6;&%23x7ed3;&%23x5408;' d='M6.71 1.11C7.53 1.11 8.19 1.78 8.19 2.59L8.19 4.41C8.36 4.35 8.53 4.31 8.71 4.31C9.34 4.31 9.88 4.7 10.09 5.25C10.28 5.16 10.49 5.11 10.71 5.11C11.34 5.11 11.88 5.5 12.09 6.05C12.28 5.96 12.49 5.91 12.71 5.91C13.53 5.91 14.19 6.58 14.19 7.39L14.19 10.19C14.19 10.22 14.19 10.25 14.19 10.28C14.19 10.31 14.19 10.34 14.19 10.37L14.19 11.19C14.19 13.23 12.55 14.87 10.51 14.87L9.3 14.87C7.9 14.87 6.56 14.32 5.57 13.33L2.23 10C1.65 9.42 1.65 8.49 2.23 7.91C2.81 7.33 3.75 7.33 4.34 7.93L5.23 8.93L5.24 2.59C5.24 1.78 5.9 1.11 6.71 1.11ZM6.71 2.08C6.43 2.08 6.19 2.31 6.19 2.59L6.19 10.19C6.19 10.23 6.19 10.26 6.18 10.3C6.17 10.33 6.14 10.41 6.13 10.44C6.12 10.45 6.08 10.51 6.07 10.51L5.99 10.58C5.99 10.59 5.92 10.63 5.91 10.63C5.89 10.64 5.81 10.66 5.79 10.67C5.77 10.67 5.74 10.67 5.71 10.67C5.7 10.67 5.65 10.67 5.64 10.67C5.58 10.66 5.52 10.64 5.47 10.61L5.4 10.55L3.64 8.59C3.44 8.39 3.11 8.39 2.91 8.59C2.71 8.79 2.71 9.12 2.91 9.32L6.25 12.65C7.06 13.46 8.16 13.91 9.3 13.91L10.51 13.91C12.02 13.91 13.24 12.7 13.24 11.19L13.24 10.37C13.24 10.34 13.24 10.31 13.24 10.28C13.24 10.25 13.24 10.22 13.24 10.19L13.24 7.39C13.24 7.11 13 6.87 12.71 6.87C12.45 6.87 12.23 7.07 12.2 7.32L12.19 7.39L12.19 9.39C12.19 9.66 11.98 9.87 11.71 9.87C11.45 9.87 11.24 9.66 11.24 9.39L11.24 6.59C11.24 6.31 11 6.08 10.71 6.08C10.45 6.08 10.23 6.27 10.2 6.52L10.19 6.59L10.19 8.59C10.19 8.86 9.98 9.08 9.71 9.08C9.45 9.08 9.24 8.86 9.24 8.59L9.24 5.79C9.24 5.51 9 5.27 8.71 5.27C8.45 5.27 8.23 5.47 8.2 5.72L8.19 5.79L8.19 8C8.19 8.26 7.98 8.47 7.71 8.47C7.45 8.47 7.24 8.26 7.24 8L7.24 2.59C7.24 2.31 7 2.08 6.71 2.08Z' fill='currentColor' fill-opacity='1.000000' fill-rule='nonzero'/%3E%3C/g%3E%3C/g%3E%3C/svg%3E");-webkit-mask:var(--v) no-repeat;mask:var(--v) no-repeat;-webkit-mask-size:100% 100%;mask-size:100% 100%;background-color:currentColor;}"`;

exports[`测试生成css的正确性 2`] = `".ci-fg{background:url("data:image/svg+xml;utf8,%3Csvg width='72.000000' height='72.000000' viewBox='0 0 72 72' xmlns='http://www.w3.org/2000/svg'%3E%3Cg fill='none'%3E%3Cg clip-path='url(%23prefix__clip7_1988)'%3E%3Cpath id='prefix__&%23x77e9;&%23x5f62;&%23x5907;&%23x4efd;-19' d='M16 0L56 0C64.8 0 72 7.2 72 16L72 56C72 64.8 64.8 72 56 72L16 72C7.2 72 0 64.8 0 56L0 16C0 7.2 7.2 0 16 0Z' fill='%237DCC29' fill-opacity='1.000000' fill-rule='evenodd'/%3E%3Cpath id='prefix__&%23x5f62;&%23x72b6;&%23x7ed3;&%23x5408;' d='M24.01 53.98L47.27 54L47.57 54C54.46 53.83 60 47.91 60 40.63C60 33.73 55.02 28.05 48.63 27.34C47 21.92 42.18 18 36.49 18C30.82 18 26.02 21.89 24.37 27.26C17.5 27.47 12 33.37 12 40.63C12 47.76 17.31 53.58 24.01 53.98ZM46.7 34.5C49.38 34.5 51.5 36.72 51.5 39.5C51.5 42.2 49.49 44.38 46.92 44.5L46.7 44.5L42 44.5C41.42 44.5 41 44.08 41 43.5C41 42.96 41.37 42.56 41.88 42.51L42 42.5L46.7 42.5C48.21 42.5 49.5 41.13 49.5 39.5C49.5 37.91 48.3 36.59 46.86 36.5L46.7 36.5L36.3 36.5C34.79 36.5 33.5 37.87 33.5 39.5C33.5 40.08 33.08 40.5 32.5 40.5C31.92 40.5 31.5 40.08 31.5 39.5C31.5 36.8 33.5 34.62 36.09 34.5L36.31 34.5L46.7 34.5ZM20.5 39.5C20.5 42.28 22.62 44.5 25.3 44.5L35.7 44.5L35.91 44.5C38.5 44.38 40.5 42.2 40.5 39.5C40.5 38.91 40.09 38.5 39.5 38.5C38.93 38.5 38.5 38.91 38.5 39.5C38.5 41.13 37.22 42.5 35.71 42.5L25.3 42.5L25.14 42.49C23.7 42.4 22.5 41.07 22.5 39.5C22.5 37.87 23.79 36.5 25.3 36.5L30 36.5L30.12 36.49C30.63 36.43 31 36.03 31 35.5C31 34.91 30.58 34.5 30 34.5L25.3 34.5L25.08 34.5C22.51 34.62 20.5 36.8 20.5 39.5Z' fill='%23FFFFFF' fill-opacity='1.000000' fill-rule='evenodd'/%3E%3C/g%3E%3C/g%3E%3C/svg%3E") no-repeat;background-size:100% 100%;background-color:transparent;}"`;
